clients/spring-boot-starter-camunda-sdk/src/main/java/io/camunda/spring/client/configuration/CredentialsProviderConfiguration.java

```diff
@@ -39,6 +41,34 @@ public class CredentialsProviderConfiguration {
   @ConditionalOnMissingBean
   public CredentialsProvider camundaClientCredentialsProvider(
       final CamundaClientProperties camundaClientProperties) {
+    if (camundaClientProperties.getMode() == ClientMode.basic) {
+      return buildBasicAuthCredentialsProvider(camundaClientProperties);
+    } else {
+      return buildOAuthCredentialsProvider(camundaClientProperties);
```

Comment:
We could use a switch - case logic here for the client mode:

saas, self-managed: OAuth
basic: BasicAuch
default: Noop